addons/website/models/res_users.py

```diff
@@ -90,3 +90,11 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_pre_authenticate_sudo.access_token = user_partner.id
                 visitor_pre_authenticate_sudo._update_visitor_last_visit()
         return uid
+
+    @api.constrains('groups_id')
+    def _check_one_user_type(self):
+        # If the user becomes internal, remove website_id.
+        super()._check_one_user_type()
+        for user in self:
+            if user._is_internal() and user.website_id:
+                user.website_id = False
```

Comment:
As this method is really a check, we hooked ourselves on it for convenience, but if we want to be correct, we should just raise a ValidationError here, telling the people to remove the website_id from the user before turning it into internal user or moving forward with whatever change they were doing